### ISSUE TITLE
tried to apply to react native project

### ISSUE THREAD

**Issue by philipheinser** [alexmckenley/react-flow-vite-template#1]
@alexmckenley I tried to copy the configs and apply it to our react native project and get an error from prettier if I use `hook` in a file.

```
["ERROR" - 19:40:01] Error formatting document.
["ERROR" - 19:40:01] Missing visitor keys for 'HookDeclaration'.
Error: Missing visitor keys for 'HookDeclaration'.
```

**Comment by alexmckenley** [comment]
I believe this error indicates that prettier is not using the "hermes" parser plugin for some reason. Double check to make sure you have both the `prettier-plugin-hermes-parser` plugin enabled and the `"parser": "hermes"` option configured as well: https://github.com/alexmckenley/react-flow-vite-template/blob/main/.prettierrc#L7-L14

Also, we've [recently merged](https://github.com/prettier/prettier/pull/16191) support for component syntax into the upstream prettier repo, so you could also try pulling the latest version, which _I believe_ should not require any special configuration.

**Comment by philipheinser** [comment]
@alexmckenley yes I saw this config and added it already will try the route to use the prettier repo

```javascript
module.exports = {
  bracketSpacing: true,
  bracketSameLine: true,
  singleQuote: true,
  trailingComma: 'all',
  arrowParens: 'avoid',
  plugins: ['prettier-plugin-hermes-parser'],
  overrides: [
    {
      files: ['*.js', '*.jsx', '*.flow'],
      options: {
        parser: 'hermes',
      },
    },
  ],
};

```
